packages/@aws-cdk/assert/test/test.have-output.ts

```diff
@@ -0,0 +1,163 @@
+  },
+  'haveOutput should assert true when output with correct name, export name and value is provided'(test: Test) {
+    expect(synthStack).to(haveOutput({
+      outputName: 'TestOutput',
+      exportName: 'TestOutputExportName',
+      outputValue: {
+        'Fn::GetAtt': [
+          'SomeResource',
+          'Arn'
+        ]
+      }
+    }));
+    test.done();
+  },
+  'haveOutput should assert false when output with correct name and export name and incorrect value is provided'(test: Test) {
+    expect(synthStack).notTo(haveOutput({
+      outputName: 'TestOutput',
+      exportName: 'TestOutputExportName',
+      outputValue: 'SomeWrongValue'
+    }));
+    test.done();
+  },
+  'haveOutput should assert true when output with correct export name and value is provided'(test: Test) {
+    expect(synthStack).to(haveOutput({
+      exportName: 'TestOutputExportName',
+      outputValue: {
+        'Fn::GetAtt': [
+          'SomeResource',
+          'Arn'
+        ]
+      }
+    }));
+    test.done();
+  },
+  'haveOutput should assert false when output with correct export name and incorrect value is provided'(test: Test) {
+    expect(synthStack).notTo(haveOutput({
+      exportName: 'TestOutputExportName',
+      outputValue: 'WrongValue'
+    }));
+    test.done();
+  },
+  'haveOutput should assert true when output with correct output name and value is provided'(test: Test) {
+    expect(synthStack).to(haveOutput({
+      outputName: 'TestOutput',
+      outputValue: {
+        'Fn::GetAtt': [
+          'SomeResource',
+          'Arn'
+        ]
+      }
+    }));
+    test.done();
+  },
+  'haveOutput should assert false when output with correct output name and incorrect value is provided'(test: Test) {
+    expect(synthStack).notTo(haveOutput({
+      outputName: 'TestOutput',
+      outputValue: 'WrongValue'
+    }));
+    test.done();
+  },
+  'haveOutput should assert false when asserting against noOutputStack'(test: Test) {
+    expect(noOutputStack).notTo(haveOutput({
+      outputName: 'TestOutputName',
+      exportName: 'TestExportName',
+      outputValue: 'TestOutputValue'
+    }));
+    test.done();
+  },
+  'haveOutput should throw Error when none of outputName and exportName is provided'(test: Test) {
+    test.throws(() => {
+      expect(synthStack).to(haveOutput({
+        outputValue: 'SomeValue'
+      }));
+    });
+    test.done();
+  },
+};
+
+function mkStack(template: any): cxapi.CloudFormationStackArtifact {
+  const templateFileName = 'test-have-output-template.json';
+  const stackName = 'test-have-output';
+  const assembly = new cxapi.CloudAssemblyBuilder();
+  assembly.addArtifact(stackName, {
+    type: cxapi.ArtifactType.AWS_CLOUDFORMATION_STACK,
+    environment: cxapi.EnvironmentUtils.format('123456789', 'ap-southeast-2'),
+    properties: {
+      templateFile: templateFileName
+    }
+  });
+
+  writeFileSync(join(assembly.outdir, templateFileName), JSON.stringify(template));
```

Comment:
I'm somewhat concerned we end up writing to disk but never clean up behind ourselves...